.github/workflows/xahau-ga-nix.yml

```diff
@@ -174,6 +174,21 @@ jobs:
           with open(os.environ['GITHUB_OUTPUT'], 'a') as f:
               f.write(f"matrix={output}\n")
 
+  apt-cache:
+    runs-on: [self-hosted, generic, 20.04]
+    container:
+      image: ubuntu:24.04
+      volumes:
+        - /home/runner/.apt-cache:/var/cache/apt
+    steps:
+      - name: Install build dependencies to save apt cache
+        run: |
+          apt-get update
+          apt-get install -y software-properties-common
+          add-apt-repository ppa:ubuntu-toolchain-r/test -y
+          apt-get update
+          apt-get install -y python3 python-is-python3 pipx cmake ninja-build ccache perl libsqlite3-dev
+
   build:
     needs: matrix-setup
```

Comment:
Is this to share apt cache between the multiple matrix entries ? 
Are the cache operations atomic ? 
Or do you need "needs: [matrix-setup, apt-cache] ? "